syntax= "proto3";
package monitor.proto;

import "google/protobuf/empty.proto";
import "cpu_load.proto";
import "cpu_softirq.proto";
import "cpu_stat.proto";

message MonitorInfo{
    string name = 1;
    repeated SoftIrq soft_irq = 4;
    CpuLoad cpu_load = 5;
    repeated CpuStat cpu_stat = 6;

}

service GrpcManager{
    rpc SetMonitorInfo(MonitorInfo) returns(google.protobuf.Empty){}

    rpc GetMonitorInfo(google.protobuf.Empty) returns (MonitorInfo){}
}

